control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2875,9 +2876,32 @@ func (r *HostedControlPlaneReconciler) reconcileClusterVersionOperator(ctx conte
 		}
 	}
 
+	if r.EnableCVOManagementClusterMetricsAccess {
```

Comment:
if this gets turned off, should there be logic for cleaning up the associated resources?

Reply:
At the moment, it shouldn't dynamically change during a HyperShift install lifecycle. I'll check the codebase for precedents.

Reply:
Oh, now I fully understand your point. I am checking.

Reply:
The created resources are being now removed in the new commit when `EnableCVOManagementClusterMetricsAccess` gets switched off.

Per commit: 

```
Remove the created objects when EnableCVOManagementClusterMetricsAccess
is switched off. Reconcile for the CVO the default ServiceAccount
"default". An empty ServiceAccountName in CVO's deployment would not
overwrite a non-empty ServiceAccountName due to the defaulting and
CreateOrUpdate logic.
```

---

However, the removal of resources in HyperShift is not quite clear to me.

For example, running `hypershift instal ... --rhobs-monitoring=false` creates the `servicemonitor.monitoring.coreos.com` objects as expected.

But after immediately running `hypershift instal ... --rhobs-monitoring` and waiting for changes to be applied, the old `servicemonitor.monitoring.rhobs` objects are still present, and new `servicemonitor.monitoring.coreos.com` are also present. I am not sure if this is an oversight or an invalid use case.

Nonetheless, CVO changes, which give more freedom to the CVO in the management cluster introduced in this PR, are removed if the flag is switched off.

Reply:
Hello, @enxebre :wave: 

Should we implement the logic to delete the associated resources in case the `EnableCVOManagementClusterMetricsAccess` gets disabled? Is this even a valid use case? I was trying to find some precedents in the codebase, but I am still not sure. I would appreciate your feedback.

Reply:
I think of this feature as a one-shot install choice which serves as the basis for the driving use case. Disabling it is not supported, granular per HC choice is not supported.

Reply:
The new commit https://github.com/openshift/hypershift/commit/0097f51dd8dd3557db6b66008a6ce285dce9a76d reverted the changes.

Marking this discussion as resolved.